doc/connectivity/bluetooth/bluetooth-ctlr-arch.rst

```diff
@@ -135,6 +135,117 @@ Execution Priorities
 
 - LLL is vendor ISR, ULL is Mayfly ISR concept, Host is kernel thread.
 
+Link Layer Control Procedures
+*****************************
+
+Following is a brief fly in on the main concepts in the implementation of
+control procedures handling in ULL.
+
++Three execution contexts
+=========================
+
+- HCI/LLCP API
+   * Miscellaneous procedure initiation API
+   * From ull_llcp.c::ull_cp_<proc>() for initiating local procedures
+   * Interface with running procedures, local and remote
+
+- lll_prepare context to drive ull_cp_run()
+   * LLCP main state machine entry/tick'er
+   * From ull_peripheral.c/ull_central.c::ticker_cb via ull_conn_llcp()
+
+- rx_demux context to drive ull_cp_tx_ack() and ull_cp_rx()
+   * LLCP tx ack handling and PDU reception
+   * From ull_conn.c::ull_conn_rx()
+   * Handles passing PDUs into running procedures as well as possibly initiating remote procedures
+
+- Not a normal execution context, but for unit testing: Emulated versions of rx_demux/prepare context used in unit tests - testing ONLY procedure PDU flow
+   * event_prepare()/event_done() helpers emulating LLL prepare/done flow
+   * lt_rx()/lt_tx() 'lower tester' emulation of rx/tx
+   * ut_rx_node() 'upper tester' emulation of notification flow handling
+   * Bunch of helpers to generate and parse PDUs, as well as misc. mocked ull_stuff()
```

Comment:
Wouldn't this section be better suited together with the other "unit test concepts"?